### PR TITLE
[IMP] speed improvement, no double None in _fields_triggers

### PR DESCRIPTION
**Pull request by fpodoo**
Speed up invoice creation by 5% allowing the same field being several times in a depends.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by fpodoo** [comment]
robodoo r+

**Comment by fpodoo** [comment]
robodoo r+

**Comment by fpodoo** [comment]
robodoo rebase-merge

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -1900,6 +1900,24 @@ def post(self):
...
+            if not move.partner_id: continue
+            if move.type.startswith('out_'):
+                field='customer_rank'
+            elif move.type.startswith('in_'):
+                field='supplier_rank'
+            else:
+                continue
+            try:
+                with self.env.cr.savepoint():
+                    self.env.cr.execute("UPDATE res_partner SET "+field+"="+field+"+1 WHERE ID=%s FOR UPDATE NOWAIT", (move.partner_id.id,))
```
Comment (fmdl) [inline addons/account/models/account_move.py]: `self.env.cr.execute("UPDATE res_partner SET "{field}"="{field}"+1 WHERE ID=%s FOR UPDATE NOWAIT".format(field=field), (move.partner_id.id,))`

File: addons/account/models/account_move.py [hunk 2]
```diff
@@ -1900,6 +1900,24 @@ def post(self):
                 # installing Accounting- with bank statements)
                 move.company_id.account_bank_reconciliation_start = move.date
 
+        for move in self:
```
Comment (fmdl) [inline addons/account/models/account_move.py]: `for move in self.filtered('partner_id'):`

**Comment by fpodoo** [comment]
merged in saas-12.5

**Pull request closed without merging**